File: programmers/월간 코드 챌린지 시즌 1/쿼드압축 후 개수 세기.js
function solution(arr) {
  const answer = [0, 0];
  const l = arr.length;
  function quadTree(r, c, l) {
    const n = arr[r][c];
    for (let i = r; i < r + l; i++) {
      for (let j = c; j < c + l; j++) {
        if (arr[i][j] != n) {
          l = parseInt(l / 2);
          quadTree(r, c, l);
          quadTree(r, c + l, l);
          quadTree(r + l, c, l);
          quadTree(r + l, c + l, l);
          return;
        }
      }
    }
    answer[n]++;
  }
  quadTree(0, 0, l);
  return answer;
}

File: programmers/월간 코드 챌린지 시즌 1/쿼드압축 후 개수 세기.py
def solution(arr):

    def quad_tree(r, c, l):
        n = arr[r][c]
        for i in range(r, r + l):
            for j in range(c, c + l):
                if arr[i][j] != n:
                    l //= 2
                    quad_tree(r, c, l)
                    quad_tree(r, c + l, l)
                    quad_tree(r + l, c, l)
                    quad_tree(r + l, c + l, l)
                    return
        answer[n] += 1

    answer = [0, 0]
    l = len(arr)
    quad_tree(0, 0, l)
    return answer
